frontend/src/search/SearchResult.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import { Link, useSearchParams } from "react-router-dom";
+import { getUsers } from "../user/user-service";
+import User from "../user/user";
+import { useGlobalContext } from "../auth/GlobalContext";
+import { format } from "date-fns";
+
+const SearchResult: React.FC = () => {
+  const [searchParams] = useSearchParams();
+  const { refresh } = useGlobalContext();
+  const [results, setResults] = useState<User[]>([]);
+
+  useEffect(() => {
+    const resultHandler = async () => {
+      const query = searchParams.get("q");
+      if (!query) return;
+
+      const res = await getUsers();
+      setResults(res.filter((User) => User.displayName.includes?.(query)));
```

Comment:
Variable `User` should not be capitalized. Also, I don't think you can put a question mark after the `.includes?`. If you did this to check if displayName exists before you run .includes, change to `displayName?.includes(query)`

Reply:
React seems to think otherwise. Program breaks if I remove the ? after .includes.